infra/modules/grafana-agent/main.tf

```diff
@@ -55,3 +55,64 @@ resource "helm_release" "grafana_agent" {
   VALUES
   ]
 }
+
+resource "helm_release" "grafana-k8s-monitoring" {
+  name       = "grafana-k8s-monitoring"
+  repository = "https://grafana.github.io/helm-charts"
+  chart      = "k8s-monitoring"
+  version    = "0.1.4"
+  namespace  = kubernetes_namespace.grafana-agent.metadata[0].name
```

Comment:
Why not define a namespace using locals and leave helm provider manage it for us?
Probably we don't need of this kubernetes_namespace resource.